src/int.rs

```diff
@@ -0,0 +1,470 @@
+    pub const BYTES: usize = Uint::<LIMBS>::BYTES;
+
+    /// The number of limbs used on this platform.
+    pub const LIMBS: usize = LIMBS;
+
+    /// Const-friendly [`Int`] constructor.
+    pub const fn new(limbs: [Limb; LIMBS]) -> Self {
+        Self(Uint::new(limbs))
+    }
+
+    /// Const-friendly [`Int`] constructor.
+    /// Note: interprets the `value` as an `Int`. For a proper conversion,
+    /// see [`Int::new_from_sign_and_magnitude`].
+    pub const fn new_from_uint(value: Uint<LIMBS>) -> Self {
+        Self(value)
+    }
+
+    /// Construct new [`Int`] from a sign and magnitude.
+    /// Returns `None` when the magnitude does not fit in an [`Int<LIMBS>`].
+    pub fn new_from_sign_and_magnitude(
+        is_negative: Choice,
+        magnitude: Uint<LIMBS>,
+    ) -> CtOption<Self> {
+        CtOption::new(
+            Self(magnitude).negate_if_unsafe(is_negative).0,
+            !magnitude.ct_gt(&Int::MAX.0) | is_negative & magnitude.ct_eq(&Int::MIN.0),
+        )
+    }
+
+    /// Create an [`Int`] from an array of [`Word`]s (i.e. word-sized unsigned
+    /// integers).
+    #[inline]
+    pub const fn from_words(arr: [Word; LIMBS]) -> Self {
+        Self(Uint::from_words(arr))
+    }
+
+    /// Create an array of [`Word`]s (i.e. word-sized unsigned integers) from
+    /// an [`Int`].
+    #[inline]
+    pub const fn to_words(self) -> [Word; LIMBS] {
+        self.0.to_words()
+    }
+
+    /// Borrow the inner limbs as an array of [`Word`]s.
+    pub const fn as_words(&self) -> &[Word; LIMBS] {
+        self.0.as_words()
+    }
+
+    /// Borrow the inner limbs as a mutable array of [`Word`]s.
+    pub fn as_words_mut(&mut self) -> &mut [Word; LIMBS] {
+        self.0.as_words_mut()
+    }
+
+    /// Borrow the limbs of this [`Int`].
+    pub const fn as_limbs(&self) -> &[Limb; LIMBS] {
+        self.0.as_limbs()
+    }
+
+    /// Borrow the limbs of this [`Int`] mutably.
+    pub fn as_limbs_mut(&mut self) -> &mut [Limb; LIMBS] {
+        self.0.as_limbs_mut()
+    }
+
+    /// Convert this [`Int`] into its inner limbs.
+    pub const fn to_limbs(self) -> [Limb; LIMBS] {
+        self.0.to_limbs()
+    }
+
+    /// Convert to a [`NonZero<Int<LIMBS>>`].
+    ///
+    /// Returns some if the original value is non-zero, and false otherwise.
+    pub const fn to_nz(self) -> ConstCtOption<NonZero<Self>> {
+        ConstCtOption::new(NonZero(self), self.0.is_nonzero())
+    }
+
+    /// Convert to a [`Odd<Int<LIMBS>>`].
+    ///
+    /// Returns some if the original value is odd, and false otherwise.
+    pub const fn to_odd(self) -> ConstCtOption<Odd<Self>> {
+        ConstCtOption::new(Odd(self), self.0.is_odd())
+    }
+
+    /// Get the sign bit of this [`Int`] as `Choice`.
+    pub fn sign_bit(&self) -> Choice {
+        ConstChoice::from_word_msb(self.0.to_words()[LIMBS - 1]).into()
+    }
+
+    /// View the data in this type as an [`Uint`] instead.
+    pub const fn as_uint(&self) -> Uint<LIMBS> {
```

Comment:
usually as_* methods return borrow value (i.e. in this case &Uint<LIMBS>.
So either rename it to to_uint or change return type.

Reply:
btw. here's the same story as transmute discussion here: https://github.com/RustCrypto/crypto-bigint/pull/695#discussion_r1803457240